src/hotspot/cpu/aarch64/assembler_aarch64.hpp

```diff
@@ -3161,6 +3175,23 @@ template<typename R, typename... Rx>
     _pmull(Vd, Ta, Vn, Vm, Tb);
   }
 
+  //Vector by element variant of UMULL
+  void umullv(FloatRegister Vd, SIMD_Arrangement Ta, FloatRegister Vn,
+               SIMD_Arrangement Tb, FloatRegister Vm, SIMD_RegVariant Ts, int lane) {
+    assert(Tb == T8B || Tb == T4H || Tb == T2S, "umullv assumes T8B, T4H, or T2S as the Tb size specifier");
+    assert(Ts == H || Ts == S, "umullv assumes H or S as the RegVariant for Ts");
+    assert(Ts == H ? Tb == T4H : Tb == T2S, "umullv assumes Tb is T4H when Ts is H");
+    _umullv(Vd, Ta, Vn, Tb, Vm, Ts, lane);
+  }
+
+  void umull2v(FloatRegister Vd, SIMD_Arrangement Ta, FloatRegister Vn,
+               SIMD_Arrangement Tb, FloatRegister Vm, SIMD_RegVariant Ts, int lane) {
+    assert(Tb == T16B || Tb == T8H || Tb == T4S, "umull2v assumes T16B, T8H, or T4S as the size specifier");
+    assert(Ts == H || Ts == S, "umull2v assumes H or S as the RegVariant for Ts");
+    assert(Ts == H ? Tb == T8H : Tb == T4S, "umull2v assumes Tb is T8H when Ts is H");
+    _umullv(Vd, Ta, Vn, Tb, Vm, Ts, lane);
+  }
+
```

Comment:
These seem to be essentially identical except for one assertion.

Suggestion: as elsewhere in class Assembler,` #define INSN` for the common pattern, then use it.